### ISSUE TITLE
[Feature Request]: Support Newpipe Subscription Imports (.json files)

### ISSUE THREAD

**Issue by longestmt** [yuliskov/SmartTube#4260]
### Describe the feature you'd like to request

The title pretty much says it. I don't want to use a google account when browsing Youtube. That is, among other reasons, a big part of why I use Tubular (e.g. Newpipe). It would be great if Smarttube was able to consume Newpipe subscription imports because I have over 100 channels that I follow. 

### Describe the solution you'd like

Consume the .json export files generated by Newpipe that contain user subscriptions. (No account required)

### Describe alternatives you've considered

[Piped.video](https://piped.video/) account integration

Edit: I always like to mention that I appreciate the work that went into this app. I just installed it on my Shield and played around with it. It definitely has potential to replace Tubular for me.

**Comment by optio50** [comment]
Yes please!
This is what keeps me from using this app exclusively. I have no desire to be coupled to a google account or add 100 channels manually. 
Can someone explain why this is not standard? (local subscriptions).
I can subscribe / view / play / add channels in what appears to be a local mode. However, when the host device reboots or the app is killed all channels / subs are gone. The good news is a backup can be restored but its a pain to add all those channel / subs.  Using a Newpipe export would be nice.

**Comment by yuliskov** [comment]
This feature has already been added into Subscription groups context menu.

**Comment by optio50** [comment]
> This feature has already been added into Subscription groups context menu.

Indeed, I just found that, thanks. However, the channel Icon and channel name are missing on the channels after the import (sidebar "Channels"). App restart and or reboot did not resolve the issue.
In addition, may i make a location suggestion about this particular setting. Make it a context menu item of the "Channels" on the sidebar. and maybe make it a default context menu item?  

Thanks for your work. I enjoy your app. -)

**Comment by optio50** [comment]
Maybe im am not understanding this setting correctly. I assumed the newpipe json import would  import the channel into the channel section. it would seem it only updates the videos in the subscription section, no channels are added. If thats the case, IMO, its only half of the equation.

**Comment by longestmt** [comment]
> This feature has already been added into Subscription groups context menu.

I will test tomorrow.

**Comment by longestmt** [comment]
> This feature has already been added into Subscription groups context menu.

I dont see a way to import the Newpipe json. If the functionality exists it is not findable. I suggest adding import/export to the settings page.

Per the other users feedback this should subscribe the app (without a signed in account) to the channels included in the json.

**Comment by longestmt** [comment]
Another update...

I finally found the file import function in the subscriptions section by long pressing. However when I tried to import my json file I get the following error "Callable returned null"

**Comment by yuliskov** [comment]
@longestmt post the json file here

**Comment by optio50** [comment]
I have experimented with this setting quite a number of times. IMO its frustrating and I believe it has undesirable behavior.
Again this is just my opinion and I have no idea what most people expect from this option, but here is how I think it would work.

If you started fresh with no channels or subscriptions (or even adding to existing).
It would be nice to long press on "Channels" or "Subscriptions", context menu pops up and select import (new pipe or freetube .json .....etc).
Result would be, all channels from the .json imported into the "Existing" channel section.
Existing "Subscriptions" (Not a new added sub category called "pinned subscriptions") would also reflect this new change.

The current behavior is somewhat random. Sometimes channels are added, sometimes not. If they are added they never have the channel name or icon in the "Channels" section.

**Comment by yuliskov** [comment]
@optio50 got it. Thanks

**Comment by optio50** [comment]
Confirmed now working in Beta 25.74

**Comment by cprados** [comment]
Hello. Thanks for this great project. Could someone please describe were the import json file from newpipe can be found in last version. I tried long pressing on different menus and sectios without success. Thanks!

**Comment by yuliskov** [comment]
@cprados it can be found subs groups ctx menu (should be explicitly enabled).

**Comment by longestmt** [comment]
> @longestmt post the json file here

I can confirm it's working in the latest release. THANKS!!

I agree with other users, however, that the import functionality should be more explicit. The context menu is pretty unintuitive. 

That being said I'm very pleased with my new default android tv youtube app. 😎

**Comment by cprados** [comment]
Thanks. Enabling add/delete channel from subscription group in general settings and long pressing on some existing channel I could acess the import from newpipe functionality.

**Comment by DaFlowah** [comment]
So, in order to import subscriptions:
1. Enable Subcriptions groups in Settings>General>Context menu
2. Open a random channel, keep pressed on a random video of such channel
3. Add/remove from a subscription group

Thanks @yuliskov and @cprados, it's been pretty difficult to find 🤣

**Comment by conkerts** [comment]
The more I read here,, the more confused I get.
I have the latest version 25.87 and I have channels and subscriptions on the left side/menu (and manually added some channels, tho it looks like in @optio50 screenshot with the black bars instead of the channel names ...).

And I just can't find the import option anywhere ...

Why not put just it in settings under Backup/Restore ?

**Comment by yuliskov** [comment]
@conkerts ok. I'll put it there.
